rest/resource.go

```diff
@@ -0,0 +1,53 @@
+package rest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/rs/rest-layer/resource"
+	"github.com/rs/rest-layer/schema/query"
+)
+
+type restResource struct {
+	*resource.Resource
+}
+
+// GetSubResource implements query.Resource interface.
+func (r restResource) Find(ctx context.Context, query *query.Query) ([]map[string]interface{}, error) {
+	itemList, err := r.Resource.Find(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	payloads := make([]map[string]interface{}, 0, len(itemList.Items))
+	for _, i := range itemList.Items {
+		payloads = append(payloads, i.Payload)
+	}
+	return payloads, nil
+}
+
+// GetSubResource implements query.Resource interface.
+func (r restResource) MultiGet(ctx context.Context, ids []interface{}) ([]map[string]interface{}, error) {
+	items, err := r.Resource.MultiGet(ctx, ids)
+	if err != nil {
+		return nil, err
+	}
+	payloads := make([]map[string]interface{}, 0, len(items))
+	for _, i := range items {
+		payloads = append(payloads, i.Payload)
+	}
+	return payloads, nil
+}
+
+// GetSubResource implements query.Resource interface.
+func (r restResource) GetSubResource(ctx context.Context, path string) (query.Resource, error) {
```

Comment:
Rename to SubResource.

It's more idiomatic in Go to drop "Get" from getter methods. Or to be more accurate, it's more idiomatic to let names be as short as they can be while still making perfect sense. Just "SubResource" is shorter, but still reads equally well, so therefore it's preferred.

- https://golang.org/doc/effective_go.html#Getters

This does not mean that the _word_ Get is banned in Go (It's perfectly well used when referring to e.g. the HTTP GET Method), just that it serves no purpose in most getter methods.